segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -315,8 +314,9 @@ protected void doStart() {
         // We are started and ready to accept requests when DurableLog starts. All other (secondary) services
         // are not required for accepting new operations and can still start in the background.
         delayedStart.thenComposeAsync(v -> {
-                    if (this.storage instanceof ChunkedSegmentStorage) {
-                        return ((ChunkedSegmentStorage) this.storage).finishBootstrap();
+                    val chunkedSegmentStorage = ChunkedSegmentStorage.getReference(this.storage);
```

Comment:
is there any log indicating slow storage would be initialized?

Reply:
Yes we do have a warning that indicates slow mode is enabled.
https://github.com/pravega/pravega/blob/master/segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/StorageLoader.java#L58-L60

However there is no log on every call. (to avoid flooding logs)